lib/src/main/java/graphql/nadel/validation/NadelHydrationArgumentValidation.kt

```diff
@@ -0,0 +1,288 @@
+                parent: NadelServiceSchemaElement,
+                overallField: GraphQLFieldDefinition,
+                remoteArg: RemoteArgumentDefinition,
+                hydration: UnderlyingServiceHydration,
+                actorFieldName: String
+        ): NadelSchemaValidationError? {
+            var hydrationSourceFieldInnerType: GraphQLType = hydrationSourceFieldType.unwrapNonNull().unwrapOne()
+            var actorFieldArgInnerType: GraphQLType = actorFieldArgType.unwrapNonNull().unwrapOne()
+            val errorExists = getHydrationInputErrors(
+                    hydrationSourceFieldInnerType,
+                    actorFieldArgInnerType,
+                    parent,
+                    overallField,
+                    remoteArg,
+                    hydration,
+                    actorFieldName
+            ) != null
+            if (errorExists) {
+                return NadelSchemaValidationError.IncompatibleHydrationArgumentType(
+                        parent,
+                        overallField,
+                        remoteArg,
+                        hydrationSourceFieldType,
+                        actorFieldArgType,
+                        actorFieldName
+                )
+            }
+            return null
+        }
+
+        private fun validateInputObjectArg(
+                hydrationSourceFieldType: GraphQLObjectType,
+                actorFieldArgType: GraphQLInputObjectType,
+                parent: NadelServiceSchemaElement,
+                overallField: GraphQLFieldDefinition,
+                remoteArg: RemoteArgumentDefinition,
+                hydration: UnderlyingServiceHydration,
+                actorFieldName: String
+        ): NadelSchemaValidationError? {
+            for (actorInnerField in actorFieldArgType.fields) {
+                val actorInnerFieldName = actorInnerField.name
+                val actorInnerFieldType = actorInnerField.type
+                val hydrationType = hydrationSourceFieldType.getField(actorInnerField.name)?.type
+                if (hydrationType == null) {
+                    if (actorInnerFieldType.isNonNull) {
+                        return NadelSchemaValidationError.MissingFieldInHydratedInputObject(
+                                parent,
+                                overallField,
+                                remoteArg,
+                                actorInnerFieldName,
+                                actorFieldName
+                        )
+                    }
+                } else {
+                    val thisFieldHasError = getHydrationInputErrors(
+                            hydrationType,
+                            actorInnerFieldType,
+                            parent,
+                            overallField,
+                            remoteArg,
+                            hydration,
+                            actorFieldName
+                    ) != null
+                    if (thisFieldHasError) {
```

Comment:
Same here, why aren't we returning the error from `getHydrationInputErrors` if it exists?

Reply:
similar to the list case, didnt want the error message to say that some random scalar type is not assignable if its wrapped an object (or wrapped multiple layers deep in objects/lists). 

We are effectively transforming the error so the error message can just say the object type that is incompatible, which i think is more useful